backend-processing.ts

```diff
@@ -302,17 +317,19 @@ export default class BackendProcessing {
   }
 
   //returns a promise boolean representing if the operation was successful
-  retrieveFundsFromDatabase(): Promise<DisplayFund[]> {
+  retrieveFundsFromDatabase(userID): Promise<DisplayFund[]> {
     return new Promise<DisplayFund[]>((resolve, reject) => {
-      database.getAllFunds().then(results => {
-        let list: DisplayFund[] = results.map(element => {
-          let fund: DisplayFund = {
-            id: element.FundID,
-            name: element.FundName
-          };
-          return fund;
+      database.getFundsUserCanSee(userID).then(availableFunds => {
```

Comment:
I'm not sure about this but I feel like `database` should be the one implementing this call function. So instead of getting everything and filtering in here, the data logic would be in the `database` object since its job is getting and filtering.
I'm not sure though, just a thought.

Mapping (`results.map`) however, is totally justifiable. If you decide to move the logic to `database` please keep the mapping here.